torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -795,21 +795,27 @@ int GatherOp::gatherAxis(int axis) const {
   return int(windowShape().size()) + axis;
 }
 
-ViewDtypeOp::ViewDtypeOp(
+ViewAsScalar::ViewAsScalar(
     IrBuilderPasskey passkey,
-    TensorView* out,
-    TensorView* in,
-    DataType dtype)
-    : Expr(passkey, ExprType::ViewDtypeOp), out_(out), in_(in), dtype_(dtype) {
+    Val* out,
+    Val* in,
+    IterDomain* vector_id,
+    Val* index)
+    : Expr(passkey, ExprType::ViewAsScalar),
+      out_(out),
+      in_(in),
+      vector_id_(vector_id),
+      index_(index) {
   addOutput(out);
   addInput(in);
 }
 
-ViewDtypeOp::ViewDtypeOp(const ViewDtypeOp* src, IrCloner* ir_cloner)
+ViewAsScalar::ViewAsScalar(const ViewAsScalar* src, IrCloner* ir_cloner)
     : Expr(src, ir_cloner),
       out_(ir_cloner->clone(src->out_)),
       in_(ir_cloner->clone(src->in_)),
-      dtype_(src->dtype()) {}
+      vector_id_(ir_cloner->clone(src->vector_id_)),
```

Comment:
I am wondering, is this correct? After the clone, would the new `vector_id_` still be identical to the last axes of `out_`?

Reply:
I think `IrCloner` takes care of it.